llvm/include/llvm/ADT/ArrayRef.h

```diff
@@ -460,8 +460,8 @@ namespace llvm {
 
     OwningArrayRef &operator=(OwningArrayRef &&Other) {
       delete[] this->data();
-      this->MutableArrayRef<T>::operator=(Other);
-      Other.MutableArrayRef<T>::operator=(MutableArrayRef<T>());
+      this->template MutableArrayRef<T>::operator=(Other);
+      Other.template MutableArrayRef<T>::operator=(MutableArrayRef<T>());
```

Comment:
This fails with Ubuntu-20.04 g++.

```
llvm/include/llvm/ADT/ArrayRef.h:463:40: error: expected ‘;’ before ‘::’ token
  463 |       this->template MutableArrayRef<T>::operator=(Other);
      |                                        ^~
      |                                        ;
llvm/include/llvm/ADT/ArrayRef.h:464:40: error: expected ‘;’ before ‘::’ token
  464 |       Other.template MutableArrayRef<T>::operator=(MutableArrayRef<T>());
      |                                        ^~
      |                                        ;
```

Reply:
```
g++ (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
```

Reply:
This appears to be a GCC bug. Requesting guidance from @AaronBallman @erichkeane @zygoloid.

Reply:
Could we revert for now?

Reply:
@chapuni I dont think this is something that can be fixed in clang. See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=94799

Reply:
So, we should revert #92957 for now, unless excluding problematic host compilers would be agreed.